pkg/document/webresolver/resolvehandler.go

```diff
@@ -0,0 +1,191 @@
+
+	localResponse, err := r.orbResolver.ResolveDocument(unpublishedOrbDID)
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			return nil, ErrDocumentNotFound
+		}
+
+		return nil, fmt.Errorf("failed to resolve document for id[%s]: %w", id, err)
+	}
+
+	didWebDoc, err := transformToDIDWeb(id, localResponse.Document)
+	if err != nil {
+		return nil, err
+	}
+
+	return &document.ResolutionResult{Document: didWebDoc}, nil
+}
+
+func transformToDIDWeb(id string, doc document.Document) (document.Document, error) {
+	alsoKnownAs := getAlsoKnownAs(doc)
+
+	if !contains(alsoKnownAs, id) {
+		// TODO: is this legit error message (should we allow it even if it is not in also known as)
+		return nil, fmt.Errorf("id[%s] not found in alsoKnownAs", id)
+	}
+
+	docBytes, err := doc.Bytes()
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal document for id[%s]: %w", id, err)
+	}
+
+	// replace all occurrences of did:orb ID with did:web ID
+	didWebDocStr := strings.ReplaceAll(string(docBytes), doc.ID(), id)
+
+	var didWebDoc document.Document
+
+	err = json.Unmarshal([]byte(didWebDocStr), &didWebDoc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal document for id[%s]: %w", id, err)
+	}
+
+	return didWebDoc, nil
+}
+
+func getAlsoKnownAs(doc document.Document) []string {
+	didDoc := document.DidDocumentFromJSONLDObject(doc)
+
+	return didDoc.AlsoKnownAs()
+}
+
+func (r *ResolveHandler) getDomainAndSuffix(id string) (string, string, error) {
+	// expected value for id is 'did:web:example.com:EiCFSKLo_K03y7YrKvBqQQ_MEjA7tgVq-KZo4W28RiGCLA
+	if !strings.HasPrefix(id, didWebPrefix) {
+		return "", "", fmt.Errorf("id[%s] has to start with did:web", id)
+	}
+
+	parts := strings.Split(id, docutil.NamespaceDelimiter)
+
+	const minDIDWebParts = 4
+	if len(parts) < minDIDWebParts {
+		return "", "", fmt.Errorf("id[%s] has to have at least four parts", id)
+	}
+
+	// suffix is always the last part
+	suffix := parts[len(parts)-1]
+
+	domain, err := util.BetweenStrings(id, didWebPrefix+docutil.NamespaceDelimiter, docutil.NamespaceDelimiter+suffix)
+	if err != nil {
+		return "", "", fmt.Errorf("retrieve domain from id[%s] failed: %w", id, err)
+	}
+
+	parsedDomain, err := parseDomain(domain)
+	if err != nil {
+		return "", "", err
+	}
+
+	logger.Debugf("returning domain[%] and suffix[%] for id[%s]", domain, suffix, id)
+
+	return parsedDomain.String(), suffix, nil
+}
+
+func parseDomain(webURL string) (*url.URL, error) {
+	// parse rules are based on steps 1-4 from: https://w3c-ccg.github.io/did-method-web/#read-resolve
+	// steps 5-7 do not apply for our dynamically generated document
+	decodedPath := strings.ReplaceAll(webURL, ":", "/")
+
+	// if the domain contains a port percent decode the colon
+	decodedPathAndPort := strings.ReplaceAll(decodedPath, "%3A", ":")
+
+	// TODO: this is did:web rule - do we need to deal with http domains?
+	urlStr := "https://" + decodedPathAndPort
```

Comment:
There's a startup parameter that dictates whether we use http: enableDevMode.

Reply:
@bstasyszyn Removed, this code (check with current domain) is not required.